src/components/structures/login/Login.js

```diff
@@ -125,7 +123,7 @@ module.exports = React.createClass({displayName: 'Login',
             // logins so let's skip that for now).
             loginLogic.chooseFlow(0);
         }, function(err) {
-            self._setErrorTextFromError(err);
+            self._setStateFromError(err);
```

Comment:
my concern is that all we have done is request the flows; if it gives a 401 or a 403, then the one thing we can be sure of is that it's *not* caused by an incorrect username/password.